net/netifyd/files/netifyd.init

```diff
@@ -11,39 +11,52 @@ USE_PROCD=1
 PROG=/usr/sbin/netifyd
 
 start_netifyd() {
-	local instance enabled autoconfig internal_if external_if
+	local autoconfig enabled instance
 
 	instance="$1"
-
 	config_get_bool enabled "$instance" enabled 0
 	[ "$enabled" -eq 0 ] && return 0
 
-	[ ! -d /var/run/netifyd ] && mkdir -p /var/run/netifyd
+	source /usr/share/netifyd/functions.sh
+	load_modules
+
+	procd_open_instance
+	procd_set_param file /etc/netifyd.conf
+	procd_set_param term_timeout 20
+	procd_set_param respawn
+	procd_set_param command $PROG -R
 
 	config_get_bool autoconfig "$instance" autoconfig 1
-	config_get filter "$instance" filter
-
-	if [ "$autoconfig" -gt 0 ] ; then
-		source /usr/share/netifyd/functions.sh
-		load_modules
-		NETIFYD_AUTODETECT=yes
-		NETIFYD_OPTS=$(auto_detect_options)
-	else
-		config_get internal_if "$instance" internal_if "eth0"
-		config_get external_if "$instance" external_if "eth1"
-		NETIFYD_OPTS="-E $external_if -I $internal_if"
+
+	if [ "$autoconfig" -gt 0 ]; then
+	    NETIFYD_AUTODETECT=yes
+	    procd_append_param command "$(auto_detect_options)"
 	fi
 
+	function append_internal_if() {
+	    procd_append_param command -I "$1"
```

Comment:
Here I hit a problem I've not idea how to solve.  The example config:
list internal_if 'eth1 -F "not (udp and dst 239.255.255.250 and dst port 1900)"'
will be passed to the -I parameter as a complete field and thus interface 'eth1 -F "not (udp and dst 239.255.255.250 and dst port 1900)"' doesn't exist.

It needs to be passed as 'eth1' '-F' 'not (foo foo foo)'

Reply:
Also I'd say the in-line functions don't enhance readability - all other scripts I've seen (which isn't all of them!) define all the functions at the top of file and then everything just uses them.

Reply:
Ok...
- Removed inline functions...
- Added common function to handle (massage) interface list entries with special handling for filter expressions (seems overly complex, but...)

Can't say I'm 100% happy -- there are possible edge cases where the "parser" could go awry, but it will do the right thing for the vast majority.  Let me know if there are bugs or a better way to implement.

Reply:
Looks better, let me give it a try :-)